vignettes/Workflow.Rmd

```diff
@@ -0,0 +1,495 @@
+Although the actual lengths of fragments vary, extending the alignment
+of the short read by a fixed amount in the appropriate direction,
+depending on whether the alignment was to the positive or negative
+strand, makes it more likely that we cover the actual site of
+interest.
+
+
+It is possible to estimate the fragment length, through a variety of
+methods. There are several implemented by the
+`estimate.mean.fraglen` function. Generally, this only needs
+to be done for one sample from each experimental protocol. Here, we
+use SSISR, the default method: 
+
+```{r estimate.mean.fraglen}
+fraglen <- estimate.mean.fraglen(cstest$ctcf, method="correlation")
+fraglen[!is.na(fraglen)] 
+```
+
+Given the suggestion of $~190$ nucleotides,
+we extend all reads to be 200 bases long. This is done using the
+`resize` function, which considers the strand to determine the direction
+of extension: 
+
+```{r}
+ctcf.ext <- resize (cstest$ctcf, width = 200)
+ctcf.ext
+```
+
+
+We now have intervals for the CTCF lane that represent the original
+fragments that were precipitated.
+
+# Coverage, islands, and depth
+
+A useful summary of this information is the
+*coverage*, that is, how many times each base in the genome was
+covered by one of these intervals.
+
+```{r}
+cov.ctcf <- coverage(ctcf.ext)
+cov.ctcf
+```
+
+For efficiency, the result is stored in a run-length encoded form.
+
+The regions of interest are contiguous segments of non-zero coverage,
+also known as *islands*.
+
+```{r}
+islands <- slice(cov.ctcf, lower = 1)
+islands
+```
+
+For each island, we can compute the number of reads in the island, and
+the maximum coverage depth within that island.
+
+```{r}
+viewSums(islands)
+viewMaxs(islands)
+
+nread.tab <- table(viewSums(islands) / 200)
+depth.tab <- table(viewMaxs(islands))
+
+nread.tab[,1:10]
+depth.tab[,1:10]
+```
+
+# Processing multiple lanes
+
+Although data from one lane is often a natural analytical unit, we
+typically want to apply any procedure to all lanes. 
+Here is a simple summary function that
+computes the frequency distribution of the number of reads.
+
+```{r}
+islandReadSummary <- function(x)
+{
+    g <- resize(x, 200)
+    s <- slice(coverage(g), lower = 1)
+    tab <- table(viewSums(s) / 200)
+    df <- DataFrame(tab)
+    colnames(df) <- c("chromosome", "nread", "count")
+    df$nread <- as.integer(df$nread)
+    df
+}
+```
+
+Applying it to our test-case, we get
+
+```{r}
+head(islandReadSummary(cstest$ctcf)) 
+``` 
+
+We can now use it to summarize
+the full dataset, flattening the returned *DataFrameList* with the
+`stack` function. 
+
+```{r} 
+nread.islands <- DataFrameList(lapply(cstest, islandReadSummary)) 
+nread.islands <- stack(nread.islands, "sample") 
+nread.islands 
+```
+
+```{r}
+xyplot(log(count) ~  nread | sample, as.data.frame(nread.islands),
+       subset = (chromosome == "chr10" & nread <= 40), 
+       layout = c(1, 2), pch = 16, type = c("p", "g"))
+```
+
+```{r fig=TRUE, height=10, echo=FALSE}
```

Comment:
Let's remove `fig=TRUE` (this may be a setting only for Sweave) and replace `height` with `fig.height`. Please do this for the other plot code blocks.